File: hw/ramos/u2/hw25MongoDBConnection/java/ContactstBook/src/main/java/ec/edu/espe/contactsbook/controller/MongoConnection.java
package ec.edu.espe.contactsbook.controller;

/**
 *
 * @author Paulo Ramos
 */

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;

public class MongoConnection {
    
    private static final String URI = "mongodb+srv://Paulo:paulo2004@cluster0.9uxqgih.mongodb.net/?appName=Cluster0";
        
    private static final String DATABASE = "ContactsDB";

    private static MongoClient mongoClient = null;

    public static MongoDatabase getConnection() {
        try {
            if (mongoClient == null) {
                mongoClient = MongoClients.create(URI);               
            }
            return mongoClient.getDatabase(DATABASE);

        } catch (Exception e) {
            System.err.println("Error al conectar con MongoDB: " + e.getMessage());
            return null;
        }
    }
}

    

File: hw/ramos/u2/hw25MongoDBConnection/python/controller/MongoConnection.py
# mongo_connection.py
from pymongo import MongoClient

class MongoConnection:
    _URI = "mongodb+srv://Paulo:paulo2004@cluster0.9uxqgih.mongodb.net/?retryWrites=true&w=majority"
    _DATABASE = "ContactsDB"
    _client = None

    @staticmethod
    def get_connection():
        try:
            if MongoConnection._client is None:
                MongoConnection._client = MongoClient(MongoConnection._URI)

            return MongoConnection._client[MongoConnection._DATABASE]

        except Exception as e:
            print("Error al conectar MongoDB:", e)
            return None
